hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/ClientVersion.java

```diff
@@ -42,6 +42,9 @@ public enum ClientVersion implements ComponentVersion {
       "This client version has support for Object Store and File " +
           "System Optimized Bucket Layouts."),
 
+  ERASURE_CODING_READ_CHUNK_CORRUPTION_FIX(4,
```

Comment:
Do we need this new client version? Can we not simple check on the server that:

```
if (blockIDProto.hasReplicaIndex()) {
  // do the verify.
}
```

I suspect we add new fields in places fairly often without creating new client versions.

I wonder what the trigger for a new client version should be, verses just checking for the presence of a field in the proto?

Reply:
Yeah we want to ensure that the block is validated everytime we do a Getblock command or ReadChunk command for newer client versions. This is to ensure that the newer client doesn't have the same regression in some other flows.

Reply:
I have ensured to the best of knowledge taken care of all the existing flows. But in future, if we forget to set replicaIndex in some other request, it should fail.

Reply:
EC_REPLICA_INDEX_REQUIRED_IN_BLOCK_REQUEST sounds good to me too.